src/consumer_queue.h

```diff
@@ -52,6 +54,9 @@ class consumer_queue {
 private:
 	send_buffer_p registry_; // optional consumer registry
 	buffer_type buffer_;	 // the sample buffer
+	// used to wait for new samples
+	std::mutex lock_;
```

Comment:
Last one, just for the sake of naming consistency with other classes (and since this is technically not the lock obj but the mutex), can we call this mut_?